### PR TITLE
Platform installer v1

### PR DESCRIPTION
**Pull request by josegomez**
Fixes a bunch of issues with the installer and race conditions on the Mac.
Moves the installer to a zip system
Teaches the installer to preserve the DB file
Fixes a few UI issues (on the Mac)

### CODE REVIEW

**Comment by josegomez** [comment]
@alexhauser spent most of the day on these, I got the releases out there and they appear to work ok for the most part.
I'm sure it isn't perfect but its a Beta release. I put together this quick video of how it functions on the Mac
https://youtu.be/hMkHZ9LQZC0

We need to update the read me file write up for installing the client. The dependencies for Linux and the Mac are the same.

File: .vscode/launch.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
             "preLaunchTask": "build",
             // If you have changed target frameworks, make sure to update the program path.
             "program": "${workspaceFolder}/SQRLDotNetClientUI/bin/Debug/netcoreapp3.1/SQRLDotNetClientUI.dll",
-            "args": [],
+            "args": ["sqrl://www.grc.com/sqrl?nut=SZNWC8znDcvovhbaT8anQg"],
```
Comment (alexhauser) [inline .vscode/launch.json]: Is this on purpose, Jose, or is it a leftover from testing?

File: SQRLPlatformAwareInstaller/ViewModels/VersionSelectorViewModel.cs [hunk 2]
```diff
@@ -360,5 +360,81 @@ public void Cancel()
...
+                    String entryFileName = zipEntry.Name;
+                    // to remove the folder from the entry:
+                    //entryFileName = Path.GetFileName(entryFileName);
+                    // Optionally match entrynames against a selection list here
+                    // to skip as desired.
+                    // The unpacked length is available in the zipEntry.Size property.
+
+                    // Manipulate the output filename here as desired.
+                    var fullZipToPath = Path.Combine(outFolder, entryFileName);
+                    if(entryFileName.Equals("sqrl.db", StringComparison.OrdinalIgnoreCase) && File.Exists(entryFileName) )
```
Comment (alexhauser) [inline SQRLPlatformAwareInstaller/ViewModels/VersionSelectorViewModel.cs:394]: We should probably replace the hardcoded db name with a static readonly variable sometime in the future, not only here but everywhere else as well. Doesn't need to be now though.

**Review by alexhauser — APPROVED** [review]
Great job, Jose! 👏

**Comment by alexhauser** [comment]
@josegomez, I tried to run the installer on Windows, but it crashes right after the download. I'll have to investigate what causes the crash.

**Comment by josegomez** [comment]
> @josegomez, I tried to run the installer on Windows, but it crashes right after the download. I'll have to investigate what causes the crash.

Taking a look, worked fine for me. But... that's always the case in these situations hehe!

File: .vscode/launch.json (hunk 1 above)
Comment (josegomez) [reply .vscode/launch.json]: That was for testing on the Mac. I'll remove it.

**Comment by alexhauser** [comment]
Ok @josegomez, here are some details regarding the crash:

A `System.Security.Principal.IdentityNotMappedException` gets thrown at line 431 of `VersionSelectorViewModel.cs`, which is this call in the zip extraction routine:

```C#
ac.AddAccessRule(fileAccessRule);
```

Here are the exception details:
```
System.Security.Principal.IdentityNotMappedException
  HResult=0x80131501
  Nachricht = Some or all identity references could not be translated.
  Quelle = System.Security.Principal.Windows
  Stapelüberwachung:
   at System.Security.Principal.NTAccount.Translate(IdentityReferenceCollection sourceAccounts, Type targetType, Boolean forceSuccess)
   at System.Security.Principal.NTAccount.Translate(Type targetType)
   at System.Security.AccessControl.CommonObjectSecurity.ModifyAccess(AccessControlModification modification, AccessRule rule, Boolean& modified)
   at System.Security.AccessControl.CommonObjectSecurity.AddAccessRule(AccessRule rule)
   at System.Security.AccessControl.FileSystemSecurity.AddAccessRule(FileSystemAccessRule rule)
   at SQRLPlatformAwareInstaller.ViewModels.VersionSelectorViewModel.ExtractZipFile(String archivePath, String password, String outFolder) in C:\Users\Alex\Source\Repos\SQRLDotNetClient\SQRLPlatformAwareInstaller\ViewModels\VersionSelectorViewModel.cs:line 431
   at SQRLPlatformAwareInstaller.ViewModels.VersionSelectorViewModel.<>c__DisplayClass51_0.<InstallingOnWindows>b__0() in C:\Users\Alex\Source\Repos\SQRLDotNetClient\SQRLPlatformAwareInstaller\ViewModels\VersionSelectorViewModel.cs:line 273
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)


```

**Comment by alexhauser** [comment]
@josegomez, looking at this more closely, I guess it is the line before that is flawed:
```C#
var fileAccessRule = new FileSystemAccessRule(new NTAccount("", "Everyone"), FileSystemRights.FullControl, AccessControlType.Allow);
```
I guess hardcoding `"Everyone"` is not a good idea here!?

**Comment by josegomez** [comment]
> @josegomez, looking at this more closely, I guess it is the line before that is flawed:
> 
> ```cs
> var fileAccessRule = new FileSystemAccessRule(new NTAccount("", "Everyone"), FileSystemRights.FullControl, AccessControlType.Allow);
> ```
> 
> I guess hardcoding `"Everyone"` is not a good idea here!?

Interesting I guess "everyone" isn't the same in other systems... that's a problem haha... I'll have to re-think this maybe we can use current user instead.... stand by

**Comment by alexhauser** [comment]
Solution according to StackOverflow:

>The name that the English version of Windows uses is "Everyone".

>You can get the user regardless of language by using the WellKnownSidType.WorldSid value:

```C#
var sid = new SecurityIdentifier(WellKnownSidType.WorldSid, null);
var account = (NTAccount) sid.Translate(typeof(NTAccount)); 
```

See https://stackoverflow.com/questions/5013881/c-sharp-how-do-i-get-the-everybody-user

**Comment by josegomez** [comment]
> Solution according to StackOverflow:
> 
> > The name that the English version of Windows uses is "Everyone".
> 
> > You can get the user regardless of language by using the WellKnownSidType.WorldSid value:
> 
> ```cs
> var sid = new SecurityIdentifier(WellKnownSidType.WorldSid, null);
> var account = (NTAccount) sid.Translate(typeof(NTAccount)); 
> ```
> 
> See https://stackoverflow.com/questions/5013881/c-sharp-how-do-i-get-the-everybody-user

Wonderful thanks!

**Comment by josegomez** [comment]
@alexhauser  fixed (and updated installer.exe in release) give her a shot.

**Comment by alexhauser** [comment]
@josegomez, works like a charm now, merge away!

**Comment by josegomez** [comment]
This resolves issue #75

**Pull request merged**